client/src/components/sources/index.tsx

```diff
@@ -66,14 +61,19 @@ const SourcesView = (): React.ReactElement => {
     }
   };
 
+  const removeSourceFromCache = (
+    cache: any,
+    { data: { deleteSource } }: any
+  ): void => updateCachedSource(cache, deleteSource, true);
+
   const [deleteSource, { loading: deletingSource }] = useMutation(
     mDeleteSource,
     { update: removeSourceFromCache, onError: onError(toaster) }
   );
 
   const onSelectSource = async (source: ISelectedSource) => {
-    if (source.credential)
-      source.credential.schema = JSON.parse(source.credential.schema as string);
+    if (source.credential && typeof source.credential.schema === 'string')
```

Comment:
I thought the JSON.parse error was thrown when source.credential.schema is an empty string, isn't it ?
I think you can just do
```javascript
if (source.credential?.schema)
...
```
no ?

Reply:
ok I misunderstood the issue, let's put this in the reducer